lib/trino-filesystem-azure/src/main/java/io/trino/filesystem/azure/AzureAuthDefault.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.filesystem.azure;
+
+import com.azure.core.credential.TokenCredential;
+import com.azure.identity.DefaultAzureCredentialBuilder;
+import com.azure.storage.blob.BlobContainerClientBuilder;
+import com.azure.storage.file.datalake.DataLakeServiceClientBuilder;
+
+public class AzureAuthDefault
+        implements AzureAuth
+{
+    private final TokenCredential credential = new DefaultAzureCredentialBuilder().build();
```

Comment:
How can this auth be tested ?

Reply:
This is for parity with S3 `DefaultCredentialsProvider`, so that Azure can be enabled by default. I don't think we need to test, since that would be testing the Azure SDK rather than our code.